File: Typescript/projects/linql.core/src/lib/LinqlSearch.ts
import { LinqlExpression } from "./LinqlExpression";
import { LinqlType } from "./LinqlType";

export class LinqlSearch 
{

    public Expressions: Array<LinqlExpression> | undefined;

    constructor(public Type: LinqlType)
    {
    }

    Merge(SearchToMerge: LinqlSearch)
    {
        this.Expressions = this.Expressions?.map(r => r.Clone());
        const lastExpression = this.Expressions?.FirstOrDefault()?.GetLastExpressionInNextChain();

        if (lastExpression)
        {
            lastExpression.Next = SearchToMerge.Expressions?.FirstOrDefault()?.Next?.Clone();
        }
    }
}

File: Python/linql-core/src/linql_core/LinqlSearch.py
from .LinqlExpression import LinqlExpression
from .LinqlType import LinqlType

class LinqlSearch():
    
    Type: LinqlType
    Expressions: list[LinqlExpression] | None

    def __init__(self, Type: LinqlType):
        self.Type = Type

    def toSerializable(self) -> dict:
        jsonObject = {}
        
        jsonObject = { "Type": self.Type.toSerializable() }
        if hasattr(self, "Expressions") and len(self.Expressions) > 0:
            jsonObject["Expressions"] = list(map(lambda x: x.toSerializable(), self.Expressions))

        return jsonObject    